File: converter.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int fahrToCel(temp){
  return (temp - 32) / 1.8;
}

int celToFahr(temp){
  return temp * 1.8 + 32;
}

void main(){
  // Introduction //
  char unit[] = "Default";
  float temp = 0;
  printf("|------------------------------|\n");
  printf("|--------Temp Converter--------|\n");
  printf("|-----------C Edition----------|\n");
  printf("|------By: Matt Hambrecht------|\n");
  printf("|------------------------------|\n");
  printf("Enter the unit of measurement: ");
  scanf(" %s", &unit);
  printf("Enter the temperature: ");
  scanf(" %f", &temp);
  printf("Unit is set to %s.\n", unit);
  printf("Temperature is set to %.5f.\n", temp);
  if(unit == "f" || unit == "fahr" || unit == "fahrenheit"){
    float fahr = temp;
    float cel = fahrToCel(temp);
    printf("Celsius = %.10f\n", cel);
    printf("Fahrenheit = %.10f\n", fahr);
  }else if(unit == "c" || unit == "cel" || unit == "celsius"){
    float fahr = temp;
    float cel = celToFahr(temp);
    printf("Celsius = %.10f\n", cel);
    printf("Fahrenheit = %.10f\n", fahr);
  }else{
    printf("%s is not a valid unit.", unit);
  }
}

File: converter.lua
-- Introduction --
print("|--------Temp Converter--------|")
print("|----------Lua Edition---------|")
print("|------By: Matt Hambrecht------|")
print("|------------------------------|")
-- Ask For Starting Unit --
io.write("Enter the unit of measurement: ")
unit = io.read()
-- Ask For Temperature --
io.write("Enter the temperature: ")
temp = io.read()
-- Evaluate Data --
if unit == "f" or "fahr" or "fahrenheit" then
  fahr = temp
  cel = (fahr - 32) / 1.8
  kel = (fahr + 459.67) * (5 / 9)
  print("Kelvin = ", kel)
  print("Celsius = ", cel)
  print("Fahrenheit = ", fahr)
elseif unit == "c" or "cel" or "celsius" then
  cel = temp
  fahr = cel * 1.8 + 32
  kel = cel + 273.15
  print("Kelvin = ", kel)
  print("Celsius = ", cel)
  print("Fahrenheit = ", fahr)
elseif unit == "k" or "kel" or "kelvin" then
  kel = temp
  fahr = (kel * 1.8) - 459.67
  cel = kel - 273.15
  print("Kelvin = ", kel)
  print("Celsius = ", cel)
  print("Fahrenheit = ", fahr)
else
  print("%s is not a valid unit!\n", unit)
end
